var/spack/repos/builtin/packages/flux-core/package.py

```diff
@@ -0,0 +1,144 @@
+    version('0.10.0', 'a84a1ed53a69c805c253bc940540cbf667a059b2008fd2a6a9bb890a985ead08e88dcbba68c01567f887357306fbfded41b93cc33edfa7809955ba5ba5870284')
+
+    # Avoid the infinite symlink issue
+    # This workaround is documented in PR #3543
+    build_directory = 'spack-build'
+
+    variant('doc', default=False, description='Build flux manpages')
+    variant('cuda', default=False, description='Build dependencies with support for CUDA')
+
+    depends_on("zeromq@4.0.4:")
+    depends_on("czmq@2.2:")
+    depends_on("hwloc@1.11.1:1.99")
+    depends_on("hwloc +cuda", when='+cuda')
+    depends_on("lua@5.1:5.1.99", type=('build', 'run', 'link'))
+    depends_on("lua-luaposix")
+    depends_on("munge")
+    depends_on("libuuid")
+    depends_on("python")
+    depends_on("py-cffi", type=('build', 'run'))
+    depends_on("jansson")
+    depends_on("yaml-cpp")
+
+    # versions up to 0.8.0 uses pylint to check Flux's python binding
+    # later versions provide a configure flag and disable the check by default
+    depends_on("py-pylint", when='@:0.8.0')
+
+    depends_on("asciidoc", type='build', when="+docs")
+
+    # Need autotools when building on master:
+    depends_on("autoconf", type='build', when='@master')
+    depends_on("automake", type='build', when='@master')
+    depends_on("libtool", type='build', when='@master')
+
+    def setup(self):
+        pass
+
+    @when('@master')
+    def setup(self):
+        # Allow git-describe to get last tag so flux-version works:
+        git = which('git')
+        git('fetch', '--unshallow')
+
+    def autoreconf(self, spec, prefix):
+        self.setup()
+        if not os.path.exists('configure'):
+            # Bootstrap with autotools
+            bash = which('bash')
+            bash('./autogen.sh')
+
+    @property
+    def lua_version(self):
+        return self.spec['lua'].version.up_to(2)
+
+    @property
+    def lua_share_dir(self):
+        return os.path.join('share', 'lua', str(self.lua_version))
+
+    @property
+    def lua_lib_dir(self):
+        return os.path.join('lib', 'lua', str(self.lua_version))
+
+    def setup_environment(self, spack_env, run_env):
+        #  Ensure ./fluxometer.lua can be found during flux's make check
+        spack_env.append_path('LUA_PATH', './?.lua', separator=';')
+
+        run_env.prepend_path(
+            'LUA_PATH',
+            os.path.join(self.spec.prefix, self.lua_share_dir, '?.lua'),
+            separator=';')
+        run_env.prepend_path(
+            'LUA_CPATH',
+            os.path.join(self.spec.prefix, self.lua_lib_dir, '?.so'),
+            separator=';')
+        run_env.prepend_path(
+            'PYTHONPATH',
+            os.path.join(
+                self.spec.prefix.lib,
+                "python{}".format(self.spec['python'].version.up_to(2)),
+                "site-packages"),
+        )
+        run_env.prepend_path(
+            'FLUX_MODULE_PATH',
+            os.path.join(self.spec.prefix.lib, 'flux', 'modules'))
```

Comment:
You can actually use `self.prefix.lib.flux.modules` here, same elsewhere. No need for `spec` or `os.path.join`.

Reply:
OOOO.  That's really slick.  Thanks!

Reply:
I'm glad you like it! That's actually one of my proudest contributions to Spack (#4591). See https://github.com/spack/spack/blob/develop/lib/spack/spack/util/prefix.py#L58,L59 for the 2 lines of code that make all the magic happen!